### PR TITLE
test: replace mocha with node:test

### PR DESCRIPTION
**Pull request by ericcrosson-bitgo**
This PR replaces mocha with the Node.js native test runner[^1].

[^1]: https://nodejs.org/api/test.html

### CODE REVIEW

File: packages/superagent-wrapper/test/request.test.ts [hunk 1]
```diff
@@ -62,236 +65,264 @@ const TestRoutes = h.apiSpec({
...
-    });
-  } else {
-    const response = PostTestRoute.response[200].encode({
-      ...params,
-      baz: true,
-    });
-    res.send(response);
-  }
-});
+const createTestServer = (port: number) => {
```
Comment (bitgopatmcl) [inline packages/superagent-wrapper/test/request.test.ts:68]: If the port is hardcoded per-test now, do we even need supertest? Or can we just use express and superagent directly?

**Review by bitgopatmcl — APPROVED** [review]
Tested and it works 🎉 . Is the main change in the `superagent-wrapper` tests to instantiate a server in the before block of each section with a hardcoded port, or did I miss something else?

File: packages/superagent-wrapper/test/request.test.ts (hunk 1 above)
Comment (bitgopatmcl) [reply packages/superagent-wrapper/test/request.test.ts:68]: Ah right, supertest also provides some extra functionality on the request object

**Comment by ericcrosson-bitgo** [comment]
> Is the main change in the `superagent-wrapper` tests to instantiate a server in the before block of each section with a hardcoded port, or did I miss something else?

That's the gist of it!

**Pull request merged**